chainstate/test-suite/src/tests/pos_processing_tests.rs

```diff
@@ -1743,3 +1743,200 @@ fn pos_stake_testnet_genesis(#[case] seed: Seed) {
         .build_and_process()
         .unwrap();
 }
+
+fn mine_pos_block(
```

Comment:
Nice function, these tests require better abstractions because they've become painfull to work with. Will add a todo for myself.